src/components/ProjectPage/ProjectPage.js

```diff
@@ -59,10 +61,10 @@ class ProjectPage extends PureComponent<Props> {
   }
 
   componentWillReceiveProps(nextProps: Props) {
-    const { project } = this.props;
+    const { project, isReinstallingActive } = this.props;
     if (
       project.id !== nextProps.project.id ||
-      project.dependencies.length === 0
+      (project.dependencies.length === 0 && !isReinstallingActive)
```

Comment:
I think this is supposed to be `isReinstallingActive` not `!isReinstallingActive` right? As in, if there's no dependencies and we triggered the reinstall, then reinstall. Otherwise the way it is now, it's saying if there's no dependencies and reinstall isn't active (aka always!)... reinstall infinitely. 

Definitely the cause of the perf issues - thanks @joshwcomeau for finding it! A gif of the nastiness in action 😮

<img width="300" alt="screen shot 2018-11-11 at 8 09 31 am" src="https://user-images.githubusercontent.com/17421347/48315423-e11fb800-e58a-11e8-9e8d-d169b230bc1b.png">

<img width="400" src="https://user-images.githubusercontent.com/17421347/48315424-e67d0280-e58a-11e8-866a-c4c3662519e0.gif">

Reply:
My idea was only load if no reinstalling is currently active. I'll try the console.log later.

Reply:
I think I get what you're trying to do with that but problem with that condition is if you're in the app not clicking the reinstall button, then you're technically in that state always (you're always having no dependencies and you're always not in reinstall state), so `loadDependencyInfoFromDisk` is being triggered infinitely.

Reply:
Ah I see, I should also get that problem if I'm removing `node_modules` folder manually in explorer. I think we should add a flow so that the loading happens only once. 
Maybe we have to add a check that will stop the loading if the folder doesn't exists.

I need to check why it's looping.